File: ForumApp/app/controllers/forumController.js
angular.module("forumApp")
.controller("forumController", ['$scope','forumService','categoryService', function ($scope, forumService, categoryService) {
    
    forumService.getForums()
    .then(function (data) {
        $scope.forums = data.data;
        $scope.filteredForums = [];
        $scope.currentPage = 1;
        $scope.numPerPage = 2;
        $scope.maxSize = 2;

        $scope.$watch('currentPage + numPerPage', function () {
            var begin = (($scope.currentPage - 1) * $scope.numPerPage)
            , end = begin + $scope.numPerPage;

            $scope.filteredForums = $scope.forums.slice(begin, end);
        });
    });

    $scope.deleteForum = function (Id) {
        forumService.deleteForum(Id)
        .then(function (data) {
            forumService.getForums()
            .then(function (data) {
                $scope.forums = data.data;
            })
            .catch(function (error) {
                $scope.error = error;
            });
        })
    };

    $scope.editForum = function (forum) {
        $scope.divForumEdit = true;
        $scope.forumId = forum.ForumId;
        $scope.forumTitle = forum.Title;
        $scope.forumContents = forum.Contents;
        $scope.forumCreatedBy = forum.CreatedBy;
    };

    $scope.editForumData = function () {
        var Forum = {
            ForumId: $scope.forumId,
            Title: $scope.forumTitle,
            Contents: $scope.forumContents,
            CreatedBy: $scope.forumCreatedBy
        };
        $scope.divForumEdit = false;
        forumService.updateForum(Forum)
        .then(function () {
            forumService.getForums()
            .then(function (data) {
                $scope.forums = data.data;
            })
        })
    };

    $scope.cancel = function () {
        $scope.forumId = "";
        $scope.forumTitle = "";
        $scope.forumContents = "";
        $scope.forumCreatedBy = "";
        $scope.divForumEdit = false;
        $scope.divForumAdd = false;
    }

    $scope.addForum = function () {
        categoryService.getCategories()
        .then(function (data) {
            $scope.categories = data.data;
            $scope.divForumAdd = true;
        })
    }

    $scope.addForumData = function () {
        var Forum = {
            Title: $scope.forumTitle,
            Contents: $scope.forumContents,
            CategoryId: $scope.selectedCategory,
            CreatedBy : $scope.forumCreatedBy
        }
        $scope.divForumAdd = false;
        forumService.addForum(Forum)
        .then(function () {
            forumService.getForums()
            .then(function (data) {
                $scope.forums = data.data;
            })
        })
    }
}])

File: ForumApp/Controllers/ForumController.cs
using ForumApp.DataAccess;
using ForumApp.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace ForumApp.Controllers
{
    public class ForumController : Controller
    {
        AppDbContext context;

        public ForumController()
        {
            context = new AppDbContext();
        }

        public JsonResult GetForums(int? Id)
        {
            if(Id == null)
            {
                var forums = context.Forums.Where(x => x.IsActive);
                return Json(forums, JsonRequestBehavior.AllowGet);
            }
            else
            {
                var forums = context.Forums.Where(x => x.IsActive && x.CategoryId == Id);
                return Json(forums, JsonRequestBehavior.AllowGet);
            }
        }

        public void EditForum(Forum forum)
        {
            var forumData = context.Forums.Where(x => x.ForumId == forum.ForumId && x.IsActive).FirstOrDefault();
            forumData.Title = forum.Title;
            forumData.Contents = forum.Contents;
            forumData.CreatedBy = forum.CreatedBy;
            context.Entry(forumData).State = System.Data.Entity.EntityState.Modified;
            context.SaveChanges();
        }

        public void AddForum(Forum forum)
        {
            forum.IsActive = true;
            context.Forums.Add(forum);
            context.SaveChanges();
        }

        public void DeleteForum(int Id)
        {
            var forum = context.Forums.Where(x => x.ForumId == Id && x.IsActive).FirstOrDefault();
            forum.IsActive = false;
            context.Entry(forum).State = System.Data.Entity.EntityState.Modified;
            context.SaveChanges();
        }

        public ActionResult Index()
        {
            return View();
        }
    }
}